src/pages/RestorePassword/RestorePassword.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { NavLink, Navigate } from "react-router-dom";
+
+import { ifAccountExist } from "../../redux/store/usersReducer";
+import { createNewPassword } from "../../redux/store/usersReducer";
+
+import { ToastRestorePassword } from "../../Components/Toast";
+import Input from "../../Components/Input";
+
+import "react-toastify/dist/ReactToastify.css";
+
+import style from "./Restore.module.scss";
+
+const RestorePassword = () => {
+  const dispatch = useDispatch();
+  const [dataToGetPassword, setDataToGetPassword] = useState({});
+  const [newPassword, setNewPassword] = useState({});
+  const [exist, setExist] = useState(true);
+  const [redirectNow, setRedirectNow] = useState(false);
+  const [success, setSuccess] = useState(null);
+
+  const curUserAccountExist = useSelector(
+    (state) => state.toolkit.curUserAccount
+  );
+
+  const passwordCheck = useSelector((state) => state.toolkit.curUserName);
+
+  const curUser = useSelector((state) => state.toolkit.curUser);
+  const checkIfUserHaveCurrentAccount = (curUserAccountObject) => {
+    dispatch(ifAccountExist(curUserAccountObject));
+  };
+
+  const getNewPassword = (newPassword, curUser) => {
+    const objToCreateNewPassword = {
+      curAccount: curUser,
+      password: newPassword,
+    };
+    setExist(true);
+
+    if (newPassword.newPassword) {
+      dispatch(createNewPassword(objToCreateNewPassword));
+
+      setSuccess(true);
+      setExist(false);
+      setTimeout(() => {
+        setRedirectNow(true);
+      }, 2000);
+    } else {
+      setSuccess(false);
+    }
+  };
+  useEffect(() => {
+    setExist(true);
+  }, [exist]);
+
+  function showInputToRestorePassword(userExist) {
```

Comment:
The showInputToRestorePassword function could be improved by returning null outside of the condition rather than just omitting it.